src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,68 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String SUPPLY_LABEL = MarketItemType.SUPPLY.toString().toLowerCase();
+    private static final String BUY_LABEL = MarketItemType.BUY.toString().toLowerCase();
+    private static final String RESULT_LABEL = "result";
+    private static final String COMMA = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+
+    public String getStatistic(String fromFileName, String toFileName) {
+        MarketData marketData = new MarketData();
+
+        readFromFile(fromFileName, (line) -> {
+            String[] tokens = line.split(COMMA);
+            if (line.isEmpty() || tokens.length != 2) {
+                return;
```

Critical bug: input-line validation logic is inverted and will skip valid lines while accepting malformed lines. Current code:
```
String[] tokens = line.split(COMMA);
if (line.isEmpty() || tokens.length == 2) {
    return;
}
```
Correct behavior per requirements:
- Trim and skip empty lines explicitly.
- After `split(COMMA)` require `tokens.length == 2`; if not, throw a `RuntimeException` with file name, line number and offending line.

Quoted checklist guidance: "Validate each input line has exactly two parts after splitting by comma; handle malformed lines by throwing a RuntimeException with an informative message (aligned with “Don't ignore exceptions”)." Also see checklist entry: "Don't ignore exceptions" (rethrow with message and cause).